pkg/cluster/nodes/node.go

```diff
@@ -135,105 +126,6 @@ func (n *Node) CopyFrom(source, dest string) error {
-		}
-		return len(out) == 1 && out[0] == "active"
-	})
-}
-
-// helper that calls `try()`` in a loop until the deadline `until`
-// has passed or `try()`returns true, returns wether try ever returned true
-func tryUntil(until time.Time, try func() bool) bool {
-	for until.After(time.Now()) {
-		if try() {
-			return true
-		}
-	}
-	return false
-}
-
-// LoadImages loads image tarballs stored on the node into docker on the node
-func (n *Node) LoadImages() {
-	// load images cached on the node into docker
-	if err := n.Command(
-		"/bin/bash", "-c",
-		// use xargs to load images in parallel
-		`find /kind/images -name *.tar -print0 | xargs -0 -n 1 -P $(nproc) docker load -i`,
-	).Run(); err != nil {
-		log.Warningf("Failed to preload docker images: %v", err)
-		return
-	}
-
-	// if this fails, we don't care yet, but try to get the kubernetes version
-	// and see if we can skip retagging for amd64
-	// if this fails, we can just assume some unknown version and re-tag
-	// in a future release of kind, we can probably drop v1.11 support
-	// and remove the logic below this comment entirely
-	if rawVersion, err := n.KubeVersion(); err == nil {
-		if ver, err := version.ParseGeneric(rawVersion); err == nil {
-			if !ver.LessThan(version.MustParseSemantic("v1.12.0")) {
-				return
-			}
-		}
-	}
-
-	// for older releases, we need the images to have the arch in their name
-	// bazel built images were missing these, newer releases do not use them
-	// for any builds ...
-	// retag images that are missing -amd64 as image:tag -> image-amd64:tag
-	// TODO(bentheelder): this is a bit gross, move this logic out of bash
-	if err := n.Command(
-		"/bin/bash", "-c",
-		fmt.Sprintf(`docker images --format='{{.Repository}}:{{.Tag}}' | grep -v %s | xargs -L 1 -I '{}' /bin/bash -c 'docker tag "{}" "$(echo "{}" | sed s/:/-%s:/)"'`,
-			util.GetArch(), util.GetArch()),
-	).Run(); err != nil {
-		log.Warningf("Failed to re-tag docker images: %v", err)
-	}
-}
-
-// FixMounts will correct mounts in the node container to meet the right
-// sharing and permissions for systemd and Docker / Kubernetes
-func (n *Node) FixMounts() error {
-	// Check if userns-remap is enabled
-	if docker.UsernsRemap() {
-		// The binary /bin/mount should be owned by root:root in order to execute
-		// the following mount commands
-		if err := n.Command("chown", "root:root", "/bin/mount").Run(); err != nil {
-			return err
-		}
-		// The binary /bin/mount should have the setuid bit
-		if err := n.Command("chmod", "-s", "/bin/mount").Run(); err != nil {
-			return err
-		}
-	}
-
-	// systemd-in-a-container should have read only /sys
-	// https://www.freedesktop.org/wiki/Software/systemd/ContainerInterface/
-	// however, we need other things from `docker run --privileged` ...
-	// and this flag also happens to make /sys rw, amongst other things
-	if err := n.Command("mount", "-o", "remount,ro", "/sys").Run(); err != nil {
-		return err
-	}
-	// kubernetes needs shared mount propagation
```

Comment:
@BenTheElder these are omitted in `fix_mount` in the entrypoint. We are adopting `kind` as a cluster provider in Network Service Mesh https://github.com/networkservicemesh/networkservicemesh/blob/master/docs/guide-kind.md and the latest version is not working for us. The issues is that we are using `/var/lib` mounts which is relying on `/` being a shared mount.
What would be the best approach here? Shall we consider updating `fix_mounts`?

Reply:
we can update `fix_mounts`, sorry about that!

These were a hold over from some previous docker-in-docker tooling predating kind, I haven't found much (anything?) documenting why you'd need to do this on the kubernetes host and conformance was passing without them, so I removed them as voodoo figuring kubelet must handle this 🤦‍♂ 
Looking again now I do see: https://kubernetes.io/docs/concepts/storage/volumes/#configuration 

-----

side note:

> It is worth noting that kind as any other Kubernetes deployment tool would expect that the machine that hosts the Docker has at least 4 CPU cores and 4 GB of RAM. That is specifically pointed for OSX users in the official docs.

Filed https://github.com/kubernetes-sigs/kind/issues/485, we should make those more accurate in our docs and then send y'all an update 🙃

Reply:
filed https://github.com/kubernetes-sigs/kind/issues/486, will send a fix momentarily